content/docs/instrumenting/writing_clientlibs.md

```diff
@@ -250,8 +250,9 @@ Client libraries MUST NOT under any circumstances allow users to have different
 label names for the same metric for Gauge/Counter/Summary/Histogram or any
 other Collector offered by the library.
 
-If your client library does validation of metrics at collect time, it MAY also
-verify this for custom Collectors.
+While metrics from custom collectors generally should also have consistent label
+names, as there are rare valid use cases where this is okay client libraries
```

Comment:
That sentence is more complex than it should be.

Reply:
What would you suggest?

Reply:
As a general rule, metrics from custom collectors generally should also have consistent label names. As there are still rare but valid use cases where this is not the case, client libraries should not verify this. An example of such a rare case would be XXX

Last sentence optional, but I can't think of any example off-hand, so including one seems useful to me.

Reply:
Adjusted

Reply:
Where? GH only shows me one commit; the initial one.

Reply:
Forgot to push it